chain/types/message_receipt_cbor.go

```diff
@@ -0,0 +1,220 @@
+package types
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/filecoin-project/go-state-types/exitcode"
+	cbg "github.com/whyrusleeping/cbor-gen"
+	"golang.org/x/xerrors"
+)
+
+// This file contains custom CBOR serde logic to deal with the new versioned
+// MessageReceipt resulting from the introduction of actor events (FIP-0049).
+
+var lengthBufMessageReceipt = []byte{132}
+
+// MarshalCBOR implements the standard marshalling logic, but omits the
+// EventsRoot field when the version is 0.
+func (mr *MessageReceipt) MarshalCBOR(w io.Writer) error {
+	if mr == nil {
+		_, err := w.Write(cbg.CborNull)
+		return err
+	}
+
+	cw := cbg.NewCborWriter(w)
+
+	if _, err := cw.Write(lengthBufMessageReceipt); err != nil {
+		return err
+	}
+
+	// t.ExitCode (exitcode.ExitCode) (int64)
+	if mr.ExitCode >= 0 {
+		if err := cw.WriteMajorTypeHeader(cbg.MajUnsignedInt, uint64(mr.ExitCode)); err != nil {
+			return err
+		}
+	} else {
+		if err := cw.WriteMajorTypeHeader(cbg.MajNegativeInt, uint64(-mr.ExitCode-1)); err != nil {
```

Comment:
woah, we need to be able to decode this in other languages too you know...

Reply:
This was generated by cborgen -- what is the concrete problem you see?